tools/src/main/java/org/apache/kafka/tools/VerifiableConsumer.java

```diff
@@ -0,0 +1,609 @@
+            return maxOffset;
+        }
+
+    }
+
+    private static ArgumentParser argParser() {
+        ArgumentParser parser = ArgumentParsers
+                .newArgumentParser("verifiable-consumer")
+                .defaultHelp(true)
+                .description("This tool produces increasing integers to the specified topic and prints JSON metadata to stdout on each \"send\" request, making externally visible which messages have been acked and which have not.");
+
+        parser.addArgument("--topic")
+                .action(store())
+                .required(true)
+                .type(String.class)
+                .metavar("TOPIC")
+                .help("Produce messages to this topic.");
+
+        parser.addArgument("--groupId")
+                .action(store())
+                .required(true)
+                .type(String.class)
+                .metavar("GROUP_ID")
+                .help("The groupId shared among members of the consumer group");
+
+        parser.addArgument("--max-messages")
+                .action(store())
+                .required(false)
+                .type(Integer.class)
+                .setDefault(-1)
+                .metavar("MAX-MESSAGES")
+                .dest("maxMessages")
+                .help("Consume this many messages. If -1, the consumer will consume until the process is killed externally");
+
+        parser.addArgument("--session-timeout")
+                .action(store())
+                .required(false)
+                .type(Integer.class)
+                .setDefault(30000)
+                .metavar("TIMEOUT_MS")
+                .dest("sessionTimeout")
+                .help("Set the consumer's session timeout");
+
+        parser.addArgument("--verbose")
+                .action(store())
+                .required(false)
+                .setDefault(false)
+                .type(Boolean.class)
+                .metavar("VERBOSE")
+                .help("Set to false to only log rebalances and commits");
+
+        parser.addArgument("--enable-autocommit")
+                .action(store())
+                .required(false)
+                .setDefault(false)
+                .type(Boolean.class)
+                .metavar("ENABLE-AUTOCOMMIT")
+                .dest("useAutoCommit")
+                .help("Enable offset auto-commit on consumer");
+
+        parser.addArgument("--broker-list")
+                .action(store())
+                .required(true)
+                .type(String.class)
+                .metavar("HOST1:PORT1[,HOST2:PORT2[...]]")
+                .dest("brokerList")
+                .help("Comma-separated list of Kafka brokers in the form HOST1:PORT1,HOST2:PORT2,...");
+
+        parser.addArgument("--consumer.config")
+                .action(store())
+                .required(false)
+                .type(String.class)
+                .metavar("CONFIG_FILE")
+                .help("Consumer config properties file.");
```

Comment:
Maybe helpful to state that the values in these properties could be overriden by the command line arguments.